### DISCUSSION TITLE
ORA-01461: can bind a LONG value only for insert into a LONG column

### DISCUSSION THREAD

**Discussion by zeales-go** [sijms/go-ora#96] [Q&A] [answered]
when table  with two clob column

CREATE TABLE ttt (
  recid                          NUMBER(16)       NOT NULL,
  itype                          VARCHAR2(4)      NOT NULL,
  subtype                        VARCHAR2(8)      NOT NULL,
  priority                       VARCHAR2(1)      NOT NULL,
  opcode                         VARCHAR2(4)      NOT NULL,
  custid                         NUMBER(16)       NOT NULL,
  servid                         NUMBER(16)       NOT NULL,
  devno                          VARCHAR2(32)         NULL,
  optime                         DATE             NOT NULL,
  etime                          DATE                 NULL,
  serialno                       VARCHAR2(20)         NULL,
  nums                           NUMBER(16)       NOT NULL,
  bizdata                        CLOB                 NULL,
  reqdata                        CLOB                 NULL,
  status                         VARCHAR2(1)          NULL
);

sql := `INSERT INTO ttt (recid,itype,subtype,priority,opcode,custid,servid,devno,optime,etime,serialno,nums,bizdata,reqdata,status) VALUES(:0,:1,:2,:3,:4,:5,:6,:7,:8,:9,:10,:11,:12,:13,:14)`

	reqdata := "a" + env.Rpad("0", 1999, "0") + "b"

	 conn_db.Exec(sql, 1, "a", "b", "c", "d", 2, 3, "e", optime, optime, "f", 0, reqdata, reqdata, "s")

result:
ORA-01461: can bind a LONG value only for insert into a LONG column

**Comment by sijms** [comment, marked as answer]
I take your code copy & paste and no error happen
but I use `reqdata := "a" + strings.Repeat("0", 2000) + "b"`
because I cannot find `env.Rpad`
please confirm that this is the code you are using
this error occur when you try to insert large data into restricted data type like varchar2
